speaker.go

```diff
@@ -0,0 +1,60 @@
+// Copyright ©2017 The ev3go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ev3dev
+
+import (
+	"encoding/binary"
+	"fmt"
+	"os"
+	"reflect"
+)
+
+const (
+	tone = 0x02
+
+	ev_snd = 0x12
+	ev_max = 0x1f
+
+	evBufLen = (ev_max + 7) / 8
+)
+
+// Tone plays a tone at the specified frequency from the ev3 speaker.
+// If freq is zero, playing is stopped.
+func Tone(freq uint32) error {
+	ok, err := hasSound()
```

Comment:
This and the path issue mean that a value that knows about the file and the path (and retains the buffer) is worth while. As a happy consequence, do *all* these gives exactly the same tone behaviour as the example you posted in the issue (previously there was an obvious lag that was not fixed by omitting this check).